### PR TITLE
fix(core): fix query processing when PK is falsy

### PR DESCRIPTION
**Pull request by Robert-Schirmer**
Fix case of `processWhere` returning empty object when the where clause is an entity with an int pk of 0

### CODE REVIEW

File: packages/core/src/utils/QueryHelper.ts [hunk 1]
```diff
@@ -95,7 +95,11 @@ export class QueryHelper {
       QueryHelper.inlinePrimaryKeyObjects(where as Dictionary, meta, metadata);
     }
 
-    where = QueryHelper.processParams(where) || {};
+    where = QueryHelper.processParams(where);
+
+    if (!where && typeof where !== 'number') {
```
Comment (B4nan) [inline packages/core/src/utils/QueryHelper.ts:100]: this should do the same, right?
```suggestion
    if (where == null) {
```

Comment (Robert-Schirmer) [reply packages/core/src/utils/QueryHelper.ts:100]: I think empty string as pk should still return `{}`, so not same for that case

Comment (B4nan) [reply packages/core/src/utils/QueryHelper.ts:100]: rest of the codebase is using the `null` checks, so this part should do it the same way. i disagree that an empty string should be treated differently than `0`

Comment (Robert-Schirmer) [reply packages/core/src/utils/QueryHelper.ts]: Ok will need to update other test that fails with that change

Comment (Robert-Schirmer) [reply packages/core/src/utils/QueryHelper.ts]: Could do `QueryHelper.processParams(where) ?? {}` then

Comment (B4nan) [reply packages/core/src/utils/QueryHelper.ts]: yeah correct, lets do it that way

**Comment by B4nan** [comment]
thanks!

**Pull request merged**